### ISSUE TITLE
performance

### ISSUE THREAD

**Issue by Lewiscowles1986** [c9s/CLIFramework#100]
Is any aspect of the CLI Framework tested for performance?

I'm using the progress bar on a few commands and it seems the commands I'm using it on are slower than some others (one of the others creates a tree structure from about 10,000 flat objects).

I'm going to conduct my own tests at some point soon, just wanted to know if there are known problems with any components performance

**Comment by c9s** [comment]
I haven't encountered performance issue yet, but i think you can reduce the frequency of updating progressbar

**Comment by Lewiscowles1986** [comment]
Apologies, CLIFramework is contributing < 10% of additional load experienced. I forgot about incremental options and had my script query an API for 9 linked resources per flat object. I've also adjusted so progressbar is only updated in my code every .1% of records (max 1000 updates). It seems the linked records was the "pain point", but reducing update frequency also helped (lol I should have known and did some testing anyway, was mainly a quick-check-in). Once I dialled down the default it sped up significantly (from 14 hours to < 30 minutes).
